awx/ui/src/screens/Setting/shared/SharedFields.js

```diff
@@ -397,7 +397,10 @@ const InputField = ({ name, config, type = 'text', isRequired = false }) => {
 };
 InputField.propTypes = {
   name: string.isRequired,
-  config: shape({}).isRequired,
+  config: shape({}),
+};
+InputField.defaultProps = {
+  config: null,
```

Comment:
Since `config` is not passed to `InputField` when values have been manually set previously, it shouldn't be a required field. And defaultProps need to be set for optional fields. https://github.com/jsx-eslint/eslint-plugin-react/issues/1433